### PR TITLE
[SPARK-27349][SQL] Dealing with TimeVars removed in Hive 2.x

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?
`hive.stats.jdbc.timeout` and `hive.stats.retries.wait` were removed by [HIVE-12164](https://issues.apache.org/jira/browse/HIVE-12164).
This pr to deal with this change.

## How was this patch tested?

unit tests

### CODE REVIEW

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala [hunk 1]
```diff
@@ -236,7 +234,18 @@ private[spark] object HiveUtils extends Logging {
       ConfVars.SPARK_RPC_CLIENT_HANDSHAKE_TIMEOUT -> TimeUnit.MILLISECONDS
     ).map { case (confVar, unit) =>
       confVar.varname -> HiveConf.getTimeVar(hadoopConf, confVar, unit).toString
-    }.toMap
+    }
+
+    // The following configurations were removed by HIVE-12164(Hive 2.0)
+    val removedTimeVars = Seq(
+      ("hive.stats.jdbc.timeout", "30s") -> TimeUnit.SECONDS,
```
Comment (wangyum) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala]: The default value from HiveConf:
https://github.com/apache/hive/blob/release-1.2.1/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java#L1287
https://github.com/apache/hive/blob/release-1.2.1/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java#L1294

**Review by attilapiros — COMMENTED** [review]
What about extending a test with two more asserts checking the default values in case of an empty configuration?

Otherwise LGTM.

(Thanks for referencing the default values from Hive.)

**Review by gatorsmile — COMMENTED** [review]
LGTM

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala [hunk 2]
```diff
@@ -236,7 +234,18 @@ private[spark] object HiveUtils extends Logging {
       ConfVars.SPARK_RPC_CLIENT_HANDSHAKE_TIMEOUT -> TimeUnit.MILLISECONDS
     ).map { case (confVar, unit) =>
       confVar.varname -> HiveConf.getTimeVar(hadoopConf, confVar, unit).toString
-    }.toMap
+    }
+
+    // The following configurations were removed by HIVE-12164(Hive 2.0)
+    val removedTimeVars = Seq(
```
Comment (gatorsmile) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala]: -> `hardcodedTimeVars`

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveUtilsSuite.scala [hunk 3]
```diff
@@ -61,4 +62,23 @@ class HiveUtilsSuite extends QueryTest with SQLTestUtils with TestHiveSingleton
...
     }
   }
+
+  test("SPARK-27349: Dealing with TimeVars removed in Hive 2.x") {
+    // Test default value
+    val defaultConf = new Configuration
+    assert(HiveUtils.formatTimeVarsForHiveClient(defaultConf)("hive.stats.jdbc.timeout") === "30")
+    assert(HiveUtils.formatTimeVarsForHiveClient(defaultConf)("hive.stats.retries.wait") === "3000")
+
+    def testFormatTimeVarsForHiveClient(key: String, value: String, expected: Long): Unit = {
```
Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveUtilsSuite.scala]: Move this out of the test suite.

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Pull request closed without merging**